quickwit-pushapi/src/push_api_service.rs

```diff
@@ -18,28 +18,63 @@
 // along with this program. If not, see <http://www.gnu.org/licenses/>.
 
 use std::path::Path;
+use std::sync::Arc;
 
 use async_trait::async_trait;
 use quickwit_actors::{Actor, ActorContext, ActorExitStatus, ActorRunner, Handler, QueueCapacity};
+use quickwit_metastore::Metastore;
 use quickwit_proto::push_api::{
     CreateQueueRequest, DropQueueRequest, FetchRequest, FetchResponse, IngestRequest,
     SuggestTruncateRequest, TailRequest,
 };
 
-use crate::{iter_doc_payloads, Position, Queues};
+use crate::{iter_doc_payloads, Position, PushApiError, Queues};
 
 pub struct PushApiService {
     queues: Queues,
+    metastore: Arc<dyn Metastore>,
 }
 
 impl PushApiService {
-    pub fn with_queue_path(queue_path: &Path) -> crate::Result<Self> {
+    pub fn with_queue_path(
+        queue_path: &Path,
+        metastore: Arc<dyn Metastore>,
+    ) -> crate::Result<Self> {
         let queues = Queues::open(queue_path)?;
-        Ok(PushApiService { queues })
+        Ok(PushApiService { queues, metastore })
     }
 
-    fn ingest(&mut self, request: IngestRequest) -> crate::Result<()> {
+    async fn ingest(&mut self, request: IngestRequest) -> crate::Result<()> {
+        // Check all indexes exist.
+        let tasks = request.doc_batches.iter().map(|batch| {
+            let moved_metastore = self.metastore.clone();
+            let index_id = batch.index_id.clone();
+            tokio::spawn(async move {
```

Comment:
I don't think spawning a task is required here.... is it?

Reply:
This has to do with `async` block. I will extract it into a function to avoid this.

Reply:
rust_analyser was giving errors on this, but build worked.